core/frontend/src/components/parameter-editor/InlineParameterEditor.vue

```diff
@@ -0,0 +1,231 @@
+      v-else-if="!custom_input && param?.options"
+      v-model.number="new_value"
+      dense
+      :items="as_select_items"
+      :rules="[() => true]"
+      :indeterminate="waiting_for_param_update"
+      @change="updateVariables"
+    />
+    <v-text-field
+      v-if="
+        custom_input
+          || (param
+            && !param.options
+            && !param.bitmask
+          )
+      "
+      v-model.number="new_value"
+      dense
+      type="number"
+      :step="param.increment ?? 0.01"
+      :suffix="param.units"
+      :rules="forcing_input ? [] : [isInRange, isValidType]"
+      :loading="waiting_for_param_update"
+      @blur="updateVariables"
+    />
+
+    <v-checkbox
+      v-if="show_advanced_checkbox"
+      v-model="forcing_input"
+      dense
+      :label="'Force'"
+    />
+    <v-checkbox
+      v-if="show_custom_checkbox"
+      v-model="custom_input"
+      dense
+      :label="'Custom'"
+    />
+  </v-form>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue'
+
+import mavlink2rest from '@/libs/MAVLink2Rest'
+import autopilot_data from '@/store/autopilot'
+import Parameter from '@/types/autopilot/parameter'
+
+export default Vue.extend({
+  name: 'InlineParameterEditor',
+  props: {
+    param: {
+      type: Object as PropType<Parameter> | undefined,
+      default: undefined,
+    },
+    allowCustom: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      custom_input: false,
+      forcing_input: false,
+      // Form can't be computed correctly, so we save it's state under data
+      is_form_valid: false,
+      new_value: 0,
+      selected_bitflags: [] as number[],
+    }
+  },
+  computed: {
+    as_select_items() {
+      const entries = Object.entries(this.param?.options ?? [])
+      return entries.map(([value, name]) => ({ text: name, value: parseFloat(value), disabled: false }))
+    },
+    edited_bitmask_value(): number {
+      return this.selected_bitflags.reduce((accumulator, current) => accumulator + current, 0)
+    },
+    show_advanced_checkbox(): boolean {
+      return typeof this.isInRange(this.new_value ?? 0) === 'string'
+    },
+    show_custom_checkbox(): boolean {
+      return !!(this.param?.options || this.param?.bitmask) && this.allowCustom
+    },
+    waiting_for_param_update(): boolean {
+      return this.param.value !== this.new_value
+    },
+    param_value() {
+      return this.param?.value ?? 0
+    },
+  },
+  watch: {
+    is_form_valid(valid) {
+      if (!valid) {
+        this.forcing_input = true
+      }
+    },
+    selected_bitflags() {
+      this.new_value = this.edited_bitmask_value
+      this.updateVariables()
+    },
+    param_value() {
+      this.updateSelectedFlags()
+    },
+  },
+  mounted() {
+    this.new_value = this.param?.value ?? 0
+    this.updateVariables()
+    this.updateSelectedFlags()
+  },
+  methods: {
+    isInRange(input: number | string): boolean | string {
```

Comment:
Can't we use the parameter editor dialog ? Or break this in a common ground ?

Reply:
refactored so editordialog uses this